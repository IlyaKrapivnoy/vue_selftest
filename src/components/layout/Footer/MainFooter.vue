<template>
  <div class="container mx-auto">
    <footer class="flex justify-center border-t-[1px]">
      <div class="my-8">&copy; By Ilya Krapivnoy {{ currentYear }}</div>
    </footer>
  </div>
</template>

<script setup>
import { ref } from "vue";

const currentYear = ref(new Date().getFullYear());
</script>
