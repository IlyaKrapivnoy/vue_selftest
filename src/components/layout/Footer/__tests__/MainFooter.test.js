import { mount } from "@vue/test-utils";
import Footer from "@/components/layout/Footer/MainFooter.vue";

describe("Footer.vue", () => {
  it("displays the current year", () => {
    const wrapper = mount(Footer);

    const currentYear = new Date().getFullYear();
    expect(wrapper.text()).toContain(`by Ilya Krapivnoy ${currentYear}`);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
