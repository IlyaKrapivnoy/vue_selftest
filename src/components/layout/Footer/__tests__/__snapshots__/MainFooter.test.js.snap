// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue displays the current year 1`] = `
"<div class="container mx-auto">
  <footer class="flex justify-center border-t-[1px]">
    <div class="my-8">Â© By Ilya Krapivnoy 2024</div>
  </footer>
</div>"
`;
